swap_meet/vendor.py

```diff
@@ -1,2 +1,105 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        # if no inventory, default to empty list
+        self.inventory = inventory if inventory is not None else []
+
+    def add(self, item):
+        # always add item to inventory
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        # remove item if it exists
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+
+    # part of wave 2
+    def get_by_id(self, id):
+        # find item in inventory by its unique id
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+
+
+    # wave 3
+    # removes my_item from this Vendor inventory and adds to friend inventory
+    # removes their_item from other Vendor inventory and adds it to this inventory
+    # method returns True
+    # If vendor inventories do not contain my_item or their_item
+    # return False
```

Comment:
Every comment, like every line of code, needs to be maintained. If the project requirements change and we have to update a function, then the comments related to it also need to be revisited and updated. Consider using comments very sparingly. 

If your code is readable and self-documenting then you shouldn't need to leave comments in the codebase very often.